plugins/modules/mongodb_user.py

```diff
@@ -202,23 +210,22 @@ def user_find(client, user, db_name):
     return False
 
 
-def user_add(module, client, db_name, user, password, roles):
+def user_add(module, client, db_name, user, password, roles, use_localhost_exception=False):
     # pymongo's user_add is a _create_or_update_user so we won't know if it was changed or updated
     # without reproducing a lot of the logic in database.py of pymongo
     db = client[db_name]
 
     try:
         exists = user_find(client, user, db_name)
     except Exception as excep:
-        # We get this exception "not authorized on admin to execute command"
-        # When there auth is enabled on a new instance. loalhost Exception
-        # Should allow us to create the first user. So we assume this is the case
-        # Might be able to do something with db.getUsers() if this approach needs
-        # a rethink
-        if excep.code == 13:  # Unauthorized
+        # We get this exception: "not authorized on admin to execute command"
+        # when auth is enabled on a new instance. The loalhost exception should
+        # allow us to create the first user.
+        # Might be able to do something with db.getUsers() if this approach needs a rethink.
+        if use_localhost_exception and 'code' in excep.keys() and excep.code == 13:  # 13=Unauthorized
```

Comment:
This need to be something like...

if hasattr(excep, 'code') and excep.code == 13:

Tests are failing because there's no keys method on this object. I'm not really sure about the flow of how this use_localhost_exception is working.

Reply:
> Tests are failing because there's no keys method on this object.

doh. True.

> I'm not really sure about the flow of how this use_localhost_exception is working.

I want to explicitly handle the localhost_exception, instead of assuming that the it applies and swallowing all unauthorized errors here. So, when we expect to use the localhost exception to add the first admin, then we ignore the unauthorized exception as searches should not work in this case. But, when we do not expect to use the localhost exception, then searches should work with the provided login credentials and so unauthorized errors should be evident.

Reply:
There's got to be a better way of handling this without changing exiting behaviour and breaking existing playbooks. This is one of the oldest modules so it's probably used a fair bit more than the others.

Reply:
OK. It turns out that it really doesn't matter if this section is unauthorized even if localhost exception doesn't apply, because the unauthorized error will be raised when adding the user. So, I backed out this part.